src/qml/FeatureForm.qml

```diff
@@ -209,160 +210,249 @@ Page {
-            if ( attributeEditorLoader.status === Loader.Error )
-            {
-              source = 'editorwidgets/TextEdit.qml'
-            }
+        Label {
+          id: constraintDescriptionLabel
+          anchors {
+            left: parent.left
+            right: parent.right
+            top: fieldLabel.bottom
           }
+
+          font.pointSize: fieldLabel.font.pointSize/3*2
+          text: {
+            if ( ConstraintHardValid && ConstraintSoftValid )
+              return '';
+
+            return ConstraintDescription || '';
+          }
+          height:  !ConstraintHardValid || !ConstraintSoftValid ? undefined : 0
+          visible: !ConstraintHardValid || !ConstraintSoftValid
+
+          color: !ConstraintHardValid ? Theme.errorColor : Theme.darkGray
         }
 
-        Connections {
-          target: form
-          onAboutToSave: {
-            // it may not be implemented
-            if ( attributeEditorLoader.item.pushChanges ) {
-              attributeEditorLoader.item.pushChanges( form.model.featureModel.feature )
+        Item {
+          id: placeholder
+          height: childrenRect.height
+          anchors { left: parent.left; right: rememberCheckbox.left; top: constraintDescriptionLabel.bottom; rightMargin: 10; }
+
+          Loader {
+            id: attributeEditorLoader
+
+            height: childrenRect.height
+            anchors { left: parent.left; right: parent.right }
+
+            //disable widget if it's:
+            // - not activated in multi edit mode
+            // - not set to editable in the widget configuration
+            // - not in edit mode (ReadOnly)
+            // - a relation in an embedded form or in multi edit mode
+            property bool isEnabled: AttributeAllowEdit
+                                     && !!AttributeEditable
+                                     && form.state !== 'ReadOnly'
+                                     && !( Type === 'relation' && ( embedded || form.model.featureModel.modelMode == FeatureModel.MultiFeatureModel ) )
+            property var value: AttributeValue
+            property var config: ( EditorWidgetConfig || {} )
+            property var widget: EditorWidget
+            property var field: Field
+            property var relationId: RelationId
+            property var nmRelationId: NmRelationId
+            property var constraintHardValid: ConstraintHardValid
+            property var constraintSoftValid: ConstraintSoftValid
+            property bool constraintsHardValid: form.model.constraintsHardValid
+            property bool constraintsSoftValid: form.model.constraintsSoftValid
+            property var currentFeature: form.model.featureModel.feature
+            property var currentLayer: form.model.featureModel.currentLayer
+            property bool autoSave: qfieldSettings.autoSave
+            // TODO investigate why StringUtils are not available in ./editorwidget/*.qml files
```

Comment:
why are they not available ;-P

Reply:
That's a pre-existing comment, I've not looking into this mysterious TODO :)